widget/playground/src/utils/common.ts

```diff
@@ -17,3 +25,103 @@ export function shallowEqual<T>(
 
   return true;
 }
+
+export const tokenToString = (token: Asset) =>
+  `${token.symbol}-${token.blockchain}-${token.address ?? ''}`;
+
+export function removeDuplicates<T>(arr: T[], key?: keyof T): T[] {
+  if (!key) {
+    return Array.from(new Set(arr));
+  }
+
+  const seen = new Set<T[keyof T]>();
+  return arr.filter((item) => {
+    const identifier = item[key];
+    if (!seen.has(identifier)) {
+      seen.add(identifier);
+      return true;
+    }
+    return false;
+  });
+}
+
+export function tokensAreEqual(tokenA?: Asset, tokenB?: Asset) {
+  return (
+    tokenA?.blockchain === tokenB?.blockchain &&
+    tokenA?.symbol === tokenB?.symbol &&
+    tokenA?.address === tokenB?.address
+  );
+}
+
+export const containsText = (text: string, searchText: string) =>
+  text.toLowerCase().indexOf(searchText.toLowerCase()) > NOT_FOUND;
+
+export const filterTokens = (list: Token[], searchedFor: string) =>
+  list.filter(
+    (token) =>
+      containsText(token.symbol, searchedFor) ||
+      containsText(token.address || '', searchedFor) ||
+      containsText(token.name || '', searchedFor)
+  );
+
+export const excludedWallets = [
+  WalletTypes.STATION,
+  WalletTypes.LEAP,
+  WalletTypes.SAFE,
+  WalletTypes.MY_TON_WALLET,
+  WalletTypes.WALLET_CONNECT_2,
+  WalletTypes.BINANCE_CHAIN,
+];
+
+export const onChangeMultiSelects = (
+  value: string,
+  values: any[] | undefined,
```

Comment:
I think we could move this function and the next one to another utils file :thinking:

Reply:
onChangeMultiSelects should be removed at the end of implementation of the playground. we haven't removed unused and previous logics yet. But I transfered the next one.